app/services/import_receivers_from_csv.rb

```diff
@@ -26,19 +26,17 @@ def successful?
 
   private
 
-  attr_reader :csv, :receivers, :institution, :success
+  attr_reader :csv_path, :receivers, :institution_id, :success
 
   def log_errors(receiver)
     Rails.logger.warn("Failed to save one of the models: #{receiver.errors.full_messages.join(' | ')}")
   end
 
   def build_receivers
-    CSV.parse(csv, headers: true) do |row|
-      receiver = Receiver.new(row.to_hash.merge(institution_id: institution.id))
+    CSV.foreach(csv_path, headers: true) do |row|
+      receiver = Receiver.new(row.to_hash.merge(institution_id: institution_id))
```

Comment:
You probably want to return in `perform` immediately if the institution does not exist, maybe? Not sure if it can be done in this part.

Reply:
Should we check before starting to parse the CSV?